parquet-variant/src/builder.rs

```diff
@@ -233,29 +234,25 @@ impl ValueBuffer {
 
 #[derive(Default)]
 struct MetadataBuilder {
-    field_name_to_id: BTreeMap<String, u32>,
-    field_names: Vec<String>,
+    field_names: IndexSet<String>,
```

Comment:
it might also help here to add some clarification, like

```suggestion
    // Field Names -- field_ids are assigned in insert order
    // which is maintained by using an IndexSet
    field_names: IndexSet<String>,
```